pkg/pool-manager/pool.go

```diff
@@ -186,31 +194,78 @@ func getNextMac(currentMac net.HardwareAddr) net.HardwareAddr {
 	return currentMac
 }
 
-// Checks if the namespace of an instance is opted in for kubemacpool
-func (p *PoolManager) isInstanceOptedIn(namespaceName, mutatingWebhookConfigName, webhookName string) (bool, error) {
+// Checks if the namespace of an instance needs to managed by kubemacpool
+func (p *PoolManager) compareNamespaceSelectorToOptMode(namespaceName, mutatingWebhookConfigName, webhookName string, vmOptMode OptMode) (bool, error) {
+	var isNamespaceManaged bool
+	if vmOptMode == OptInMode {
+		// in OptInMode namespaces are not managed by default
+		isNamespaceManaged = false
+	} else if vmOptMode == OptOutMode {
+		// in OptOutMode namespaces are managed by default
+		isNamespaceManaged = true
+	} else {
+		return false, fmt.Errorf("opt-mode is not defined: %s", vmOptMode)
+	}
+
 	ns, err := p.kubeClient.CoreV1().Namespaces().Get(context.TODO(), namespaceName, metav1.GetOptions{})
 	if err != nil {
-		return false, errors.Wrapf(err, "Failed to get Namespace %s", namespaceName)
+		return isNamespaceManaged, errors.Wrapf(err, "Failed to get Namespace %s", namespaceName)
```

Comment:
This patter is contravention is general: On failure, the caller is usually supposed to ignore the value (non error type ones). The method, usually returns the "init value" and attempts to express that it should not be examined.

In this case, it will be strange to see it returning `true` and error at the same time. It make me wander if the caller is suppose to examine it even on errors (which is again uncommon).

Reply:
I agree. In case of failure I would prefer to not handle the namespace.